### PR TITLE
clarinet 0.14.1 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/clarinet.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Clarinet < Formula
+  desc "Command-line tool and runtime for the Clarity smart contract language"
+  homepage "https://github.com/hirosystems/clarinet"
+  url "https://github.com/hirosystems/clarinet/archive/v0.14.1.tar.gz"
+  sha256 "33ff6bc312f27fc450406091b67fac16be48b553b309f61538e8d0b16dbbb191"
+  license "GPL-3.0-only"
+  head "https://github.com/hirosystems/clarinet.git"
```
Comment (nandahkrishna) [inline Formula/clarinet.rb:7]:
```suggestion
  head "https://github.com/hirosystems/clarinet.git", branch: "main"
```

**Review by nandahkrishna — COMMENTED** [review]
Need to change the branch for `head`. Also, are they using a vendored `deno`?

**Comment by alebcay** [comment]
That's a good question. @CharlieC3 and @MarvinJanssen, we noticed that you have a copy of Deno living in `vendor/deno`. Any way we can use a system/existing copy of Deno instead of that?

**Comment by CharlieC3** [comment]
> That's a good question. @CharlieC3 and @MarvinJanssen, we noticed that you have a copy of Deno living in `vendor/deno`. Any way we can use a system/existing copy of Deno instead of that?

This might be a better question for @lgalabru to answer.

**Comment by lgalabru** [comment]
Yes, Deno is being vendored (instead of pulling it as a crate), because we're only using a subset of Deno, and mainly using their engine for transpiling Typescript and spawning some v8 VM.  With this approach, we can limit the number of implied dependencies, and make our command line tool lighter + faster to compile.

**Comment by alebcay** [comment]
@nandahkrishna thoughts on this? It sounds like the alternative would've been pulling in Deno as a crate anyways. It sounds like Clarinet more or less embeds Deno as a library/component, and brewed Deno acts more as a standalone framework/runtime.

**Comment by nandahkrishna** [comment]
> It sounds like the alternative would've been pulling in Deno as a crate anyways. It sounds like Clarinet more or less embeds Deno as a library/component, and brewed Deno acts more as a standalone framework/runtime.

Since this is the case, fine with me 👍🏼.

**Review by nandahkrishna — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Nice addition. 🚀

**Pull request closed without merging**